### ISSUE TITLE
Petition to collapse "*cachegroup*"  endpoints

### ISSUE THREAD

**Issue by ocket8888** [apache/trafficcontrol#2934]
# The problem statement
At the moment we have all of these endpoints for dealing with cachegroups

* /api/1.x/cachegroup/{{parameter ID}}/parameter
* /api/1.x/cachegroup_fallbacks
* /api/1.x/cachegroupparameters
* /api/1.x/cachegroups
* /api/1.x/cachegroups/{{ID}}
* /api/1.x/cachegroups/{{ID}}/deliveryservices
* /api/1.x/cachegroups/{{ID}}/parameters
* /api/1.x/cachegroups/{{ID}}/queue_updates
* /api/1.x/cachegroups/{{ID}}/unassigned_parameters
* /api/1.x/cachegroups/{{parameter ID}}/parameter_available
* /api/1.x/cachegroups_trimmed

These are inconsistently named and implemented, provide a lot of overlapping and related functionality, and are pretty difficult to look at. I think it'd be much easier to use if there were only one or two endpoints :

* /api/1.4/cachegroups
* /api/1.4/cachegroupparameters

Most of the other endpoints can be pretty trivially implemented as query parameters to these two endpoints (in fact, parameters could also be grouped under /api/1.x/cachegroups, but that might get ugly). 

## /api/1.x/cachegroup/{{parameter ID}}/parameter
As far as I can tell, this one doesn't even work - or, at least, I don't understand what it's supposed to accomplish. Using it - with any parameter ID, even ones that aren't for Cache Groups, or don't even exist, or aren't even numbers (I've tried it with `9001`, `-1` and `one` and get the same thing every time) will **ALWAYS** simply return the names and IDs of all the cache groups. So this should probably just be gotten rid of, possibly with the functionality of emitting IDs added to "trimmed"

## /api/1.x/cachegroup_fallbacks
This can be very easily added to the functionality of /api/1.x/cachegroups - just add the list of fallbacks to cachegroup objects that have fallbacks. Or, if we hate that idea, add the `fallbacks=[true|false]` query param to turn the feature on/off

## /api/1.x/cachegroupparameters
Currently, this reports the list of all (?) parameters assigned to any (?) cachegroup. I think that's fine, but maybe the `"cachegroup"` object could be an array? Would cut down on output size by a lot in environments where cachegroup params are re-used.

## /api/1.x/cachegroups
The way this works is fine, imho (apart from collapsing the other stuff into it, as discussed in other sections)

## /api/1.x/cachegroups/{{ID}}
This should really just be the `id={{ID}}` query parameter. Not much else to say about that.

## /api/1.x/cachegroups/{{ID}}/deliveryservices
This could easily be added to the functionality of `/api/1.x/cachegroups` by just adding the key `"deliveryservices"` to cachegroup objects as a list of Delivery Service IDs. Or, if we hate that idea, add the `deliveryservices=[true|false]` query param to turn the feature on/off. If the /api/1.x/cachegroups{{ID}} endpoint gets absorbed into /api/1.4/cachegroups as a query parameter, there's no real reason to limit this to a single cachegroup by ID because that can be done with a separate parameter.

## /api/1.x/cachegroups/{{ID}}/queue_updates
Unlike all the others here, I think this one shouldn't actually be absorbed into one of the proposed v1.4 cachegroup endpoints. Instead, this feels like it belongs as a query parameter to `/api/1.2/cdns/{{ID}}/queue_update`, especially since this requires you to specify a CDN anyway.

## /api/1.x/cachegroups/{{ID}}/unassigned_parameters
This one just has a confusing use-case in my opinion. In what situation would you want to know all of the parameters *not* assigned to a specific cachegroup? Isn't that just the difference of the set of all parameters assigned to a specific cachegroup from the set of all parameters? Surely the number of parameters *not* assigned to any one cachegroup is always much larger than the number of parameters that *are* assigned to it?

## /api/1.x/cachegroups/{{parameter ID}}/parameter_available
I haven't found a Traffic Ops instance that actually responds to this request with anything but a 404. Maybe I don't understand what it's supposed to do (and trust me when I say I have no idea what this would do supposing it worked), but if this doesn't exist anymore I'll omit the documentation for it in my next Docs update pull request.

## /api/1.x/cachegroups/trimmed
This one's a no-brainer, it'd be so easy to implement as the query parameter `trimmed=[true|false (default)].

# TL;DR
In conclusion, collapsing the API would make it easier to document, easier to read, and - I believe - easier to write if we do it this way in the future. Prefering query parameters to route parameters makes related functionalities easy to link, and would expose the most common functionality in the least verbose/complicated way.
